.github/workflows/trivy.yml

```diff
@@ -2,12 +2,44 @@ name: Trivy vulnerability scanner
 on:
   push:
     branches:
-      - master
+      - main
   pull_request:
 jobs:
-  build:
-    name: Build
+  build-and-scan:
+    name: Scan ${{ matrix.image }}
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: docker-build-workspace
+            image: test/workspace:latest
+            registry: test
+
+          - target: docker-build-ragengine
+            image: test/ragengine:latest
+            registry: test
+
+          - target: docker-build-adapter
+            image: test/e2e-adapter:0.0.1
+            registry: test
+
+          - target: docker-build-adapter
+            image: test/e2e-adapter2:0.0.1
+            registry: test
+
+          - target: docker-build-adapter
+            image: test/adapter-phi-3-mini-pycoder:0.0.1
+            registry: test
+
+          - target: docker-build-dataset
+            image: test/e2e-dataset:0.0.1
+            registry: test
+
+          - target: docker-build-dataset
+            image: test/e2e-dataset2:0.0.1
+            registry: test
```

Comment:
These images are not exposed to users. It's used in the e2e tests.

Reply:
I removed these and left the matrix strategy, I think it might prove useful down the track if we choose to add further images to our scans